### PR TITLE
Integrate getStoredTaskResultByRequestId in xpanse

### LINKED ISSUE #2130: Integrate  getStoredTaskResultByRequestId in xpanse
- [ ] Add a new mandatory column 'HANDLER' to the SERVICE_ORDER.
- [ ] Values for HANDLER -> TERRAFORM-LOCAL, TERRAFORM-BOOT, OPENTOFU-LOCAL, TOFU-MAKER, INTERNAL, WORKFLOW, AGENT, PLUGIN
- [ ] Update all inserts for SERVICE_ORDER to have this value. 
- [ ] Add new configuration - max.service.order.processing.duration.in.seconds=180
- [ ] Whenever there is an API call that returns 'DeployedService' or DeployedServiceDetails and when the corresponding service is stuck in DESTROYING/DEPLOYING/MODIFYING for more than the above configuration time.. (order created time > the max processing duration) and if the order handler is TERRAFORM-BOOT or TOFU-MAKER, then call the newly added service in #2079 and #2129 respectively to fetch the latest status.  If they return an exception (Already returned or not found), then set the order status to ERROR with the error message from API and set the ServiceDeploymentState to MANUAL_CLEANUP_REQUIRED. 
- [x] If API returns 204 (implemented in #2131), then return the status same as what is stored in xpanse DB

### PR DESCRIPTION
**Pull request by WangLiNaruto**
Fixed #2130

### CODE REVIEW

File: modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/DeployService.java [hunk 1]
```diff
@@ -770,4 +780,27 @@ public ServiceTemplateEntity getOrderableServiceDetailsByServiceId(UUID serviceI
         return serviceTemplateStorage.getServiceTemplateById(
                 deployedService.getServiceTemplateId());
     }
+
+    private Handler getHandler(String activeProfile, DeployerKind kind) {
+        if (StringUtils.isEmpty(activeProfile)) {
+            return Handler.INTERNAL;
```
Comment (baixinsui) [inline modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/DeployService.java:786]: The activeProfile could not be empty, this case is not right.

File: modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/DeployService.java [hunk 2]
```diff
@@ -770,4 +780,27 @@ public ServiceTemplateEntity getOrderableServiceDetailsByServiceId(UUID serviceI
...
+        } else if (kind.equals(DeployerKind.OPEN_TOFU)) {
+            if (activeProfiles.contains(Handler.TOFU_MAKER.toValue())) {
+                return Handler.TOFU_MAKER;
+            } else {
+                return Handler.OPEN_TOFU_LOCAL;
+            }
+        } else {
+            return Handler.TERRAFORM_LOCAL;
+        }
+    }
```
Comment (baixinsui) [inline modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/DeployService.java:804]: Or maybe you can set INTERNAL as the default value as the result.

**Comment by Alice1319** [comment]
LGTM

**Comment by yuanxiaojun987** [comment]
LGTM

File: modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/ServiceDetailsViewManager.java [hunk 3]
```diff
@@ -298,5 +331,133 @@ private ServiceQueryModel getServiceQueryModel(Category category, Csp csp, Strin
         return query;
     }
 
+    private void updateServiceDeploymentStateAndServiceOrderByStoredTerraformResult(
```
Comment (swaroopar) [inline modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/ServiceDetailsViewManager.java:334]: refetchDeploymentStateForMissingOrdersFromDeployers

**Review by swaroopar — CHANGES_REQUESTED** [review]

File: modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/ServiceDetailsViewManager.java (hunk 3 above)
Comment (swaroopar) [inline modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/ServiceDetailsViewManager.java:334]: better we move all this new code to a new spring bean.

File: modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/ServiceDetailsViewManager.java [hunk 4]
```diff
@@ -298,5 +331,133 @@ private ServiceQueryModel getServiceQueryModel(Category category, Csp csp, Strin
...
+    }
+
+    private void updateServiceDeploymentState(ServiceDeploymentEntity serviceDeployment,
+            ServiceOrderEntity serviceOrderEntity) {
+        if (Objects.nonNull(serviceOrderEntity)) {
+            if (Duration.between(serviceOrderEntity.getStartedTime(),
+                    OffsetDateTime.now()).getSeconds() > maxServiceOrderProcessingDuration) {
+                if (serviceOrderEntity.getHandler().equals(Handler.TERRAFORM_BOOT)) {
+                    try {
+                        ResponseEntity<TerraformResult> result = terraformBootTaskResultHelper
```
Comment (swaroopar) [inline modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/ServiceDetailsViewManager.java:396]: would prefer if we extract the response object in the terraformBootTaskResultHelper class itself and send the actual response object

File: modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/deployers/opentofu/tofumaker/TofuMakerTaskResultHelper.java [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Component;
+import org.springframework.web.client.RestClientException;
+
+/**
+ * Bean to manage task result via tofu-maker.
+ */
+@Slf4j
+@Component
+public class TofuMakerTaskResultHelper {
```
Comment (swaroopar) [inline modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/deployers/opentofu/tofumaker/TofuMakerTaskResultHelper.java:22]: TofuMakerResultRefetchManager

File: modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/deployers/terraform/terraformboot/TerraformBootTaskResultHelper.java [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Component;
+import org.springframework.web.client.RestClientException;
+
+/**
+ * Bean to manage task result via terraform-boot.
+ */
+@Slf4j
+@Component
+public class TerraformBootTaskResultHelper {
```
Comment (swaroopar) [inline modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/deployers/terraform/terraformboot/TerraformBootTaskResultHelper.java:22]: TerraformBootResultRefetchManager

**Review by swaroopar — CHANGES_REQUESTED** [review]

File: modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/DeployResultManager.java [hunk 7]
```diff
@@ -406,4 +410,62 @@ private void processRelatedWorkflowTask(ServiceOrderEntity serviceOrder) {
...
+        serviceDeploymentStorage.storeAndFlush(serviceDeployment);
+    }
+
+    /**
+     * update service state and service order state by exception.
+     */
+    public void updateServiceDeploymentStateAndServiceOrder(ServiceDeploymentEntity serviceEntity,
+            ServiceOrderEntity serviceOrder, ErrorType errorType, HttpClientErrorException e) {
+        Response response = e.getResponseBodyAs(Response.class);
+        if (e.getStatusCode() == HttpStatus.BAD_REQUEST
```
Comment (swaroopar) [inline modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/DeployResultManager.java:455]: please move all this to corresponding result handlers.

File: modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/DeployResultManager.java [hunk 8]
```diff
@@ -406,4 +410,62 @@ private void processRelatedWorkflowTask(ServiceOrderEntity serviceOrder) {
...
+                && Objects.nonNull(response)
+                && response.getResultType()
+                == Response.ResultTypeEnum.RESULT_ALREADY_RETURNED_OR_REQUEST_ID_INVALID) {
```
Comment (swaroopar) [inline modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/DeployResultManager.java:458]: ensure we use the 'Response' class from generated models of deployers.

File: modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/DeployResultManager.java [hunk 9]
```diff
@@ -406,4 +410,62 @@ private void processRelatedWorkflowTask(ServiceOrderEntity serviceOrder) {
...
+            serviceEntity
+                    .setServiceDeploymentState(ServiceDeploymentState.MANUAL_CLEANUP_REQUIRED);
+            serviceOrder.setTaskStatus(TaskStatus.FAILED);
+            serviceOrder.setErrorResponse(ErrorResponse.errorResponse(errorType,
+                    List.of(e.getMessage())));
+            serviceDeploymentStorage.storeAndFlush(serviceEntity);
+            serviceOrderStorage.storeAndFlush(serviceOrder);
+        } else {
+            log.error(String.format("Modify service failed. orderId %s, error %s ",
```
Comment (swaroopar) [inline modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/DeployResultManager.java:467]: but here we are not modifying anything know?

**Review by swaroopar — CHANGES_REQUESTED** [review]

**Review by swaroopar — APPROVED** [review]
LGTM

**Pull request merged**